app/com/github/pedrovgs/dotto/core/DottoApp.scala

```diff
@@ -0,0 +1,26 @@
+package com.github.pedrovgs.dotto.core
+
+import cats.free.Free
+import com.github.pedrovgs.dotto.core.algebra.dotto._
+
+/**
+  * Declares the two programs Dotto implements.
+  */
+object DottoApp {
+
+  def enqueueMessageProgram: Message => Free[Instruction, Message] = {
+    messageToEnqueue: Message =>
+      for {
```

Comment:
If you have a single step the for comprehension it's kind of pointless, try just invoking the function or a simple flatMap. Also all public APIs including this `def` should explicitly provide a return type. The compiler will do less work instead of attempting to infer the types.